src/main/java/nyx/Task.java

```diff
@@ -0,0 +1,54 @@
+package nyx;
+
+/**
+ * Represents a task.
+ * Contains information about the task and whether it is marked as done.
+ */
+public abstract class Task {
+    private final String content;
+    private boolean isDone;
+
+    /**
+     * Constructor for Task. (For invocation by subclass constructors)
+     * @param content Description of the task.
+     * @param isDone Indicator to indicate whether the task is done.
+     */
+    public Task(String content, boolean isDone) {
+        this.content = content;
+        this.isDone = isDone;
+    }
+
+    private String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    protected int getStatusInt() {
+        return isDone ? 1 : 0;
+    }
+
+    String getContent() {
```

Comment:
Perhaps adding an access modifier here would make thing method easier to read!

Reply:
i think the author intended it to be package private here, but i might be mistaken. @g4ryy is this method being used elsewhere in your package?

Reply:
Hi @rohit0718, thanks for pointing this out! Indeed, my intention is to make this method package private since it is not used outside the package.